app/vmselect/promql/eval.go

```diff
@@ -814,7 +815,14 @@ func evalRollupFunc(qt *querytracer.Tracer, ec *EvalConfig, funcName string, rf
 			Err: fmt.Errorf("`@` modifier must return a single series; it returns %d series instead", len(tssAt)),
 		}
 	}
-	atTimestamp := int64(tssAt[0].Values[0] * 1000)
+	atValue := math.NaN()
+	for _, v := range tssAt[0].Values {
+		if !math.IsNaN(v) {
+			atValue = v
+			break
+		}
+	}
+	atTimestamp := int64(atValue * 1000)
```

Comment:
Probably, it brings us back to the original problem, if atValue is Nan, it will result into -maxInt64 timestamp. Should we check it and return some error instead?

Reply:
Added a check to return an error if all values are NaN